sw/device/lib/runtime/irq_vector.S

```diff
@@ -0,0 +1,46 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+.extern irq_type_scratch_area
+.extern irq_internal_dispatch
+
+#define HANDLER_STUB(label, type)     \
+  label:                              \
+    sw s0, -4(sp);                    \
+    sw s1, -8(sp);                    \
+    li s0, type;                      \
+    sw s0, irq_type_scratch_area, s1; \
```

Comment:
was something supposed to have been loaded into s1?

Reply:
No, the third argument is a scratch register in which to load the symbol offset.

Reply:
right i get that.  I just meant s1 is not taking on a value explicitly in this sequence.  Since we store s1 onto the stack, doesn't that mean we intend to overwrite its original value with something else?

Reply:
sorry, i'm clearly not a great assembly writer, sorry if i misunderstood.